xrmocap/core/estimation/mview_mperson_topdown_estimator.py

```diff
@@ -0,0 +1,293 @@
+                start_idx=start_idx,
+                end_idx=end_idx,
+                img_arr=img_arr,
+                img_paths=img_paths,
+                video_paths=video_paths,
+                logger=self.logger)
+            bbox_list, keypoints2d_list = self.estimate_perception2d(
+                mview_batch_arr)
+            for sub_fidx in range(end_idx - start_idx):
+                sframe_bbox_list = []
+                sframe_keypoints2d_list = []
+                for view_idx in range(n_view):
+                    sframe_bbox_list.append(bbox_list[view_idx][sub_fidx])
+                    mframe_keypoints2d = keypoints2d_list[view_idx]
+                    keypoints2d = Keypoints(
+                        kps=mframe_keypoints2d.get_keypoints()[
+                            sub_fidx:sub_fidx + 1, ...],
+                        mask=mframe_keypoints2d.get_mask()[sub_fidx:sub_fidx +
+                                                           1, ...],
+                        convention=mframe_keypoints2d.get_convention(),
+                        logger=self.logger)
+                    n_kps = keypoints2d.get_keypoints_number()
+                    sframe_bbox_list.append(bbox_list[view_idx][sub_fidx])
+                    sframe_keypoints2d_list.append(keypoints2d)
+                association_results, predict_keypoints3d, identities = \
+                    self.associator.associate_frame(
+                        mview_img_arr=mview_batch_arr[:, sub_fidx, ...],
+                        mview_bbox2d=sframe_bbox_list,
+                        mview_keypoints2d=sframe_keypoints2d_list,
+                        affinity_type='geometry_mean'
+                    )
+                for p_idx in range(len(association_results)):
+                    identity = identities[p_idx]
+                    associate_idxs = association_results[p_idx]
+                    tri_kps2d = np.zeros(shape=(n_view, n_kps, 3))
+                    tri_mask = np.zeros(shape=(n_view, n_kps))
+                    for view_idx in range(n_view):
+                        kps2d_idx = associate_idxs[view_idx]
+                        if kps2d_idx is not None:
+                            tri_kps2d[view_idx] = \
+                                sframe_keypoints2d_list[
+                                    view_idx].get_keypoints()[
+                                        0, kps2d_idx, ...]
+                            tri_mask[view_idx] = \
+                                sframe_keypoints2d_list[
+                                    view_idx].get_mask()[
+                                        0, kps2d_idx, ...]
+                    if self.point_selectors is not None:
+                        for selector in self.point_selectors:
+                            tri_mask = selector.get_selection_mask(
+                                points=tri_kps2d, init_points_mask=tri_mask)
+                    kps3d = self.triangulator.triangulate(tri_kps2d, tri_mask)
```

Comment:
The `predict_keypoints3d` is the end result, we don't need to triangulate again.

Reply:
If we take `predict_keypoints3d`  from MVPose Associator or any other associator, point_selection is not complicated enough. In estimator, we could use more than one point_selector.